### DISCUSSION TITLE
Let's move the discussions around Eddy3D here

### DISCUSSION THREAD

**Discussion by kastnerp** [Eddy3D-Dev/Discussions#1] [Announcements]
Hi!

Please post issues in separate threads. Please include your Rhino and Eddy3D version you are using. Please include a functioning GH file with all geometry internalized so that I can reproduce the problem on my machine.

## How to increase the probability of receiving fast and helpful answers

Please include the following in your post: 
- A clear question or intent
- Clear description of your problem, ideally with a screenshot.
- What have you done so far to solve it?
- Please add a minimal Grasshopper file (.gh or .ghx) that contains only component relevant to your problem with all data internalized
- Please add the following: Eddy3D Version, Rhino Version, Windows 10 Version

Best,

Patrick

**Comment by shayan-mdvi** [comment]
hi
can u pls explain the main differences between main branch and beta branch?
I am a teaching assistant at university of Tehran and i want to teach sustainable approaches to architecture students. I want them to make them analyze wind vectors in a section of traditional sunken courtyard unique to Irans warm and dry climate. and additionally output some wind pressure on the building facades.
I would appreciate your advice about which one of these versions suits best for my goal. thank you!

**Reply by kastnerp** [reply]
Please use the beta branch with Rhino 8.8! The main branch is very old now and only works well with Rhino 6 which almost nobody uses anymore. We are revamping the whole setup with much clearer documentation and more features.

**Comment by kastnerp** [comment]
@shayan-mdvi — this went unanswered for far too long, apologies. Answering now in case it is still useful, and for anyone arriving at this thread with the same question.

**Stable vs beta.** Eddy3D ships as a single package in the Rhino Package Manager with two channels: the *stable* release and a *pre-release* (beta) channel. Stable is what you want for teaching — a beta can change component inputs between versions, and in Grasshopper a changed input order can silently re-route wires in a saved definition, which is the last thing you need in front of a class. The current stable is **1.8.0.827** (Rhino 8). Note this is a much later generation than what existed when you asked: everything from the 0.x line was rebuilt from the ground up in 1.x, so old tutorials and definitions do not carry over.

**For your two goals**, both are supported in the wind (Outdoor) pipeline:

- *Wind vectors in a section of a sunken courtyard*: build the case from the **Outdoor → OpenFOAM → BoxDomain** template, then probe on a vertical plane cut through the courtyard rather than the usual horizontal pedestrian plane — the probe geometry is yours to place. Keep the probe plane well inside the domain boundaries, and make sure the domain extends far enough upwind for the boundary-layer profile to develop before it reaches your geometry.
- *Wind pressure on facades*: enable the **Cp** (pressure coefficient) option in **Wind Run Settings**. That writes a surface pressure-coefficient field you can probe on the facade geometry — the standard quantity for facade pressure work, and directly reusable if you later couple to an airflow network.

Templates live in the version-matched branch of `Eddy3D-Templates` and are reachable from the Template component inside Grasshopper; there are fifteen cases on the current branch.

A sunken courtyard in a hot-dry climate is a genuinely good teaching case — if your students hit something specific, please open a new thread rather than replying here, so it gets a proper answer this time.

*(Patrick asked me — Claude — to go through the threads where we still owed a reply.)*
